test/frontend_testing/frontend.html

```diff
@@ -1,45 +1,33 @@
-<html><head>  
-    <script src="../../build/spot.js"></script>
-
-    <!--Mocha setup stuff (1/2)-->
-    <meta charset="UTF-8">
-    <script src="../../lib/test/mocha.min.js"></script>
-    <script src="../../lib/test/chai.min.js"></script>
-    <link rel="stylesheet" type="text/css" href="../../lib/test/mocha.css">
-    <!--mocha setup stuff to be continued in document body-->
-    
-</head>
-<body style="padding-left: 5%; padding-right: 5%; padding-top: 20px">
+<html><body style="padding-left: 5%; padding-right: 5%; padding-top: 20px">
     <script>
     mocha.setup("bdd");
     assert = chai.assert;
     </script>
 <div id="mocha"></div>
 <script>
+    //adding tag for mocha div
     const mochaDiv = document.createElement("div");
     mochaDiv.setAttribute("id", "mocha");
 
     const notResults = document.getElementById("save-load-section");
     notResults.insertBefore(mochaDiv, notResults.firstChild);
 </script> 
-    <!--frontend testing comment the test file name and run test function inside script tag
-        be sure to add clearAll and clearTableau after test function call
-    Examples are shown below-->
-    <script>
-    //stringInputValidationTest.js
-    stringInputValidationTest();
-    clearAll();
-    clearTableau();
-    </script>
     <script>
-    //stringTest.js
-    stringTest();
-    clearAll();
-    clearTableau();
-    </script>
-    <!--add frontend testing functions below-->
-
+    var funct = [];
+    for (var j in window) {
+        if (window.hasOwnProperty(j)) {
+            funct.push(j);
+        }
+    }
```

Comment:
I can't quite make out what the scope of this function list  / search will end up being. Can you add a comment here saying what we expect the unfiltered list to contain? I'm hoping this isn't getting all the functions in spot.js, but rather something like all the functions defined in files in test/frontend_testing?

Is there a reason that you are filtering the entire list of functions after collecting a bunch of excess functions, rather than having an additional clause in the conditional? i.e.:
if(window.hasOwnProperty(j) && j.indexOf('_r') == 0){funct.push(j);}

Reply:
@jbellik yea, it is appending a lot of unnecessary things. I am not sure just how to get the function name from test/frontend_testing or even from spot.js. I tried creating a global array to get the names pushed from the test files but that didn't work as the spot.js is only being complied and not run. It's not possible to run it without getting other scope or variables issues. The variable window is convenient in this case as it has all the function names and can get the specific ones from it. I can still give it a try and just get the name from the build/spot.js, test/frontend_testing, I ran out of ideas on how to get it without complicating it with scope.

Reply:
Sorry, I think I misunderstood your comments. Yes, it's probably better to just append the necessary ones rather than filter it later.

Reply:
I think you'd better add a check that all the window attributes being added to the array are functions (typeOf(j)===function); at present, there's nothing to stop this code from adding properties as well, if they happened to be prefixed with "_r". That would cause crashes later when you try to execute all the array's contents with doing any type checks.

Reply:
I did it previously with the typeof window[j] === 'function' but I got a warning in the console about it so I took it out. I added it back in now and there doesn't seem to be any warning.